cmd/tailscale/main.go

```diff
@@ -570,6 +585,46 @@ func (a *App) runBackend(ctx context.Context) error {
 	}
 }
 
+// editPrefs is called when prefs is created for the first time. It pulls
+// from the edit function channel, and makes calls to localapi to edit the
+//
+//	preferences. This ensures that edits will be made in the correct order.
+func (a *App) editPrefs(ctx context.Context, prefs ipn.PrefsView) {
+	for {
+		select {
+		case fn := <-a.editFns:
+			mp := new(ipn.MaskedPrefs)
+			fn(prefs, mp)
+
+			jsonData, err := json.Marshal(mp)
+			if err != nil {
+				log.Printf("error marshaling MaskedPrefs %v", err)
+				continue
+			}
+
+			body := bytes.NewReader(jsonData)
+			r, err := a.localAPIClient.Call(ctx, "PATCH", "prefs", body)
+			if err != nil {
+				log.Printf("localapiclient error %v", err)
+				continue
+			}
+
+			data, err := io.ReadAll(r.Body())
+			r.Body().Close()
+			if err != nil {
+				log.Printf("error reading localapi response %v", err)
+			}
+			var newPrefs ipn.PrefsView
+			err = json.Unmarshal(data, &newPrefs)
+			if err != nil {
+				log.Printf("error unmarshaling localapi response %v", err)
+			}
+			prefs = newPrefs
+		default:
```

Comment:
Having this `default` here is going to cause your for loop to spin in a tight loop.

I think what you want is just `for fn := range a.editFns {}`.

See https://go.dev/play/p/z5HWzCU93_e